### ISSUE TITLE
Catch::Session::applyCommandLine signature incorrect for non-const argv

### ISSUE THREAD

**Issue by RossBencina** [philsquared/Catch#615]
The C and C++ standards specify that the signature for `main` requires argv to be modifiable.[1][2] For example: `int main(int argc, char* argv[])`. In particular, declaring argv const is non-portable.

If `main` is declared as above, then calling `Catch::Session::applyCommandLine` produces an error in MSVC, gcc and clang.

MSVC 2015:

```
main.cpp(133): error C2664: 'int Catch::Session::applyCommandLine(int,const char *[],Catch::Session::OnUnusedOptions::DoWhat)': cannot convert argument 2 from 'char *[]' to 'const char *[]'
1>         c:\amtrunk_wd\fluent\fluenttest_testmain.cpp(133): note: Conversion loses qualifiers
```

gcc:

```
prog.cc:15:29: error: invalid conversion from 'char**' to 'const char**' [-fpermissive]
```

clang:

```
prog.cc:1:6: note: candidate function not viable: no known conversion from 'char **' to 'const char **' for 2nd argument
```

Here's a sandbox to experiment with:
http://melpon.org/wandbox/permlink/bFwLOVsdbBMkOmUh

**Proposed Fix**

Change the signature of `applyCommandLine` and `parseInto` as follows:

```
-        int applyCommandLine( int argc, char const* argv[], OnUnusedOptions::DoWhat unusedOptionBehaviour = OnUnusedOptions::Fail ) {
+        int applyCommandLine( int argc, const char* const* argv, OnUnusedOptions::DoWhat unusedOptionBehaviour = OnUnusedOptions::Fail ) {

-        std::vector<Parser::Token> parseInto( int argc, char const* argv[], ConfigT& config ) const {
+        std::vector<Parser::Token> parseInto( int argc, const char* const* argv, ConfigT& config ) const {
```

This change fixes the issue in MSVC, and should do so in gcc and clang also, based on the sandbox test above.

[1] http://stackoverflow.com/questions/2108192/what-are-the-valid-signatures-for-cs-main-function
[2] http://stackoverflow.com/questions/11216155/is-int-mainint-char-const-const-well-formed

**Comment by RossBencina** [comment]
Here's the proposed fix:
https://github.com/RossBencina/Catch/commit/18f18ae830ba95089ef0d734802292c332b25e61

**Comment by horenmar** [comment]
This has been fixed in 1c47fe023a1547e9a7a7e4efd163c098d69e38a4, so I am going to close this.
